sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -325,6 +330,15 @@ public void setAttribute(String key, AttributeValue value) {
         logger.log(Level.FINE, "Calling setAttribute() on an ended Span.");
         return;
       }
+      totalAttributeCount++;
+      if (attributes.get(key) == null && attributes.size() >= maxNumberOfAttributes) {
+        logger.log(
+            Level.FINE,
+            "Span has maximum number of attributes ("
+                + maxNumberOfAttributes
+                + "). Dropping new entries.");
```

Comment:
You could also add the span's name and the key of the dropped attribute to ease debugging here.